motionplan/stubs_nocgo.go

```diff
@@ -0,0 +1,50 @@
+//go:build no_cgo
+
+package motionplan
+
+import (
+	"math/rand"
+
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+const pathdebug = false
+
+var errNotSupported = errors.New("not supported on this build")
+var flipPose = spatialmath.NewPoseFromOrientation(&spatialmath.OrientationVectorDegrees{OZ: 1, Theta: 180})
+
+func newCBiRRTMotionPlanner(
+	fs referenceframe.FrameSystem,
+	seed *rand.Rand,
+	logger logging.Logger,
+	opt *plannerOptions,
+) (motionPlanner, error) {
+	return nil, errNotSupported
+}
+
+// newTPSpaceMotionPlanner creates a newTPSpaceMotionPlanner object with a user specified random seed.
+func newTPSpaceMotionPlanner(
+	fs referenceframe.FrameSystem,
+	seed *rand.Rand,
+	logger logging.Logger,
+	opt *plannerOptions,
+) (motionPlanner, error) {
+	return nil, errNotSupported
+}
+
+// NewRRTStarConnectMotionPlannerWithSeed creates a rrtStarConnectMotionPlanner object with a user specified random seed.
+func newRRTStarConnectMotionPlanner(
+	fs referenceframe.FrameSystem,
+	seed *rand.Rand,
+	logger logging.Logger,
+	opt *plannerOptions,
+) (motionPlanner, error) {
+	return nil, errNotSupported
+}
+
+func rectifyTPspacePath(path []node, frame referenceframe.Frame, startPose spatialmath.Pose) ([]node, error) {
```

Comment:
This collection of stubs is so odd to me. Can you describe why these are needed and why they are sufficient?

Reply:
these are functions that call newNloptSomething directly

reason they aren't needed in no_cgo mode -- the CLI is only using the client; implementations are still happening inside RDK

(and in cgo builds, the original definitions are still used; I didn't change them)